metagenomics.py

```diff
@@ -78,8 +99,42 @@ def parser_krona(parser=argparse.ArgumentParser()):
     util.cmd.attach_main(parser, krona, split_args=True)
     return parser
 
+def diamond(inBam, db, outM8, threads=1):
+    tmp_fastq = util.file.mkstempfname('.fastq')
+    tmp_fastq2 = util.file.mkstempfname('.fastq')
+    picard = tools.picard.SamToFastqTool()
+    picard_opts = {
+        'CLIPPING_ATTRIBUTE': picard.illumina_clipping_attribute,
+        'CLIPPING_ACTION': 'X'
+    }
+    picard.execute(inBam, [tmp_fastq, tmp_fastq2] , unpaired=True,
+                   picardOptions=picard.dict_to_picard_opts(picard_opts),
+                   JVMmemory=picard.jvmMemDefault)
+
+    diamond_tool = tools.diamond.Diamond()
+    diamond_tool.install()
+    tmp_alignment = util.file.mkstempfname('.daa')
+    tmp_m8 = util.file.mkstempfname('.diamond.m8')
+    diamond_tool.blastx(db, [tmp_fastq], tmp_alignment,
```

Comment:
Instead of modifying tools/picard.py, why not just do a normal SamToFastq and then cat the two outputs together to a temp file for diamond to take as input?